### PR TITLE
Iteration 4

### PR DESCRIPTION
**Pull request by Maxwell-Baird**
Modified the games_truncated to include both post season and regular season.
Updated the tests that use games_truncated  to reflect the new data
Created two methods for seasonal_stats biggest bust and biggest surprise 
Added test to check biggest bust and biggest surprise

### CODE REVIEW

File: data/games_truncated.csv [hunk 1]
```diff
@@ -20,3 +20,15 @@ game_id,season,type,date_time,away_team_id,home_team_id,away_goals,home_goals,ve
...
+2012020489,20122013,Regular Season,3/28/13,27,30,3,2,Exploria Stadium,/api/v1/venues/null
+2012020239,20122013,Regular Season,2/22/13,1,15,3,2,Providence Park,/api/v1/venues/null
+2012020587,20122013,Regular Season,4/10/13,7,52,1,2,Providence Park,/api/v1/venues/null
+2012020651,20122013,Regular Season,4/19/13,23,25,1,3,SeatGeek Stadium,/api/v1/venues/null
+2012020379,20122013,Regular Season,3/12/13,23,29,1,1,Exploria Stadium,/api/v1/venues/null
+2013020727,20132014,Regular Season,1/19/14,8,10,3,3,WakeMed Soccer Park,/api/v1/venues/null
+2013020010,20132014,Regular Season,10/4/13,18,19,2,2,Talen Energy Stadium,/api/v1/venues/null
+2013020920,20132014,Regular Season,3/4/14,29,10,2,1,WakeMed Soccer Park,/api/v1/venues/null
+2012020065,20122013,Regular Season,1/27/13,7,15,2,3,Providence Park,/api/v1/venues/null
+2012020003,20122013,Regular Season,1/19/13,16,26,3,2,Nippert Stadium,/api/v1/venues/null
```
Comment (whitneykidd) [inline data/games_truncated.csv:34]: Nice catch!

File: lib/season_stats.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 require_relative 'stats'
-
+require 'pry'
```
Comment (alex-latham) [inline lib/season_stats.rb:2]: Looks like this was accidentally left in

File: test/league_stats_test.rb [hunk 3]
```diff
@@ -140,32 +140,33 @@ def test_game_teams_stats_update_id
...
+    assert_equal "Toronto FC", @league_stats.best_defense
   end
 
   def test_returns_worst_defense
     assert_equal "Sporting Kansas City", @league_stats.worst_defense
   end
 
-  def test_returns_average_defense
-    @league_stats.average_defense
-  end
```
Comment (alex-latham) [inline test/league_stats_test.rb:170]: Good catch about this not having an assertion. We'll double-check about this method's test when we match all of our existing methods to tests near the end.

File: data/games_truncated.csv (hunk 1 above)
Comment (tylertomlinson) [reply data/games_truncated.csv:34]: agreed!

File: lib/season_stats.rb (hunk 2 above)
Comment (tylertomlinson) [reply lib/season_stats.rb:2]: There is one on runner as well we can get rid of eventually

File: lib/season_stats.rb [hunk 4]
```diff
@@ -11,6 +11,76 @@ def shot_accuracy_by_team_id(team_id)
...
+  def check_empty(away, home, wins_hash)
+    if wins_hash[away] == nil
+      wins_hash[away] = [0,0]
+    end
+    if wins_hash[home] == nil
+      wins_hash[home] = [0,0]
+    end
+    wins_hash[home][1] += 1
+    wins_hash[away][1] += 1
+  end
```
Comment (tylertomlinson) [inline lib/season_stats.rb:68]: nice methods!

File: test/game_stats_test.rb [hunk 5]
```diff
@@ -38,29 +38,29 @@ def test_returns_biggest_blowout
...
   end
 
   def test_it_can_calculate_average_goals_per_game
-    assert_equal 3.81, @game_stats.average_goals_per_game
+    assert_equal 4.0, @game_stats.average_goals_per_game
   end
 
   def test_it_can_calculate_average_goals_per_season
-    expected = {"20122013"=>3.81}
+    expected = {"20122013"=>3.97, "20132014"=>4.33}
```
Comment (tylertomlinson) [inline test/game_stats_test.rb:63]: Looks like fun

File: test/league_stats_test.rb [hunk 6]
```diff
@@ -140,32 +140,33 @@ def test_game_teams_stats_update_id
...
+                27=>2, 1=>2, 7=>5, 23=>4, 18=>2, 29=>1}
 
     assert_equal expected, @league_stats.away_id_defense_stats
   end
 
   def test_defense_helper_returns_correct_hash
-    expected = {6=>10, 3=>14, 5=>10, 16=>13, 17=>10, 8=>14, 9=>9}
+    expected = {6=>10, 3=>14, 5=>10, 16=>15, 17=>10, 8=>17, 9=>9, 25=>6, 19=>3,
+                30=>5, 15=>5, 52=>1, 29=>2, 10=>5, 26=>3, 20=>2, 27=>2, 1=>2,
+                7=>5, 23=>4, 18=>2}
```
Comment (tylertomlinson) [inline test/league_stats_test.rb:159]: FUN!

**Pull request merged**